cmake/Dependencies.cmake

```diff
@@ -149,3 +149,20 @@ IF(USE_MKLDNN)
     INCLUDE_DIRECTORIES(${MKLDNN_INCLUDE_DIR})
     LIST(APPEND SINGA_LINKER_LIBS ${MKLDNN_LIBRARIES})
 ENDIF()
+
```

Comment:
make mpi and nccl necessary only for distributed training.
we can add an optional for this in CMakefiles.